pkg/nebula/meta_client.go

```diff
@@ -368,41 +342,68 @@ func (m *metaClient) BalanceStatus(jobID int32, spaceID nebula.GraphSpaceID) err
 		Type:    meta.JobType_STATS,
 		Paras:   [][]byte{[]byte(strconv.FormatInt(int64(jobID), 10))},
 	}
-	resp, err := m.client.RunAdminJob(req)
+	resp, err := m.retryOnError(req, func(req interface{}) (interface{}, error) {
+		resp, err := m.client.RunAdminJob(req.(*meta.AdminJobReq))
+		return resp, err
+	})
 	if err != nil {
 		return err
 	}
-	if len(resp.GetResult_().JobDesc) > 0 {
-		if resp.GetResult_().JobDesc[0].Status == meta.JobStatus_FINISHED {
+	jobDesc := resp.(*meta.AdminJobResp).GetResult_().JobDesc
+	if len(jobDesc) > 0 {
+		if jobDesc[0].Status == meta.JobStatus_FINISHED {
 			return nil
 		}
 	}
 	log.Info("Balance job in progress")
 	return &utilerrors.ReconcileError{Msg: fmt.Sprintf("Balance job %d still in progress", jobID)}
 }
 
-func (m *metaClient) retryOnError(req interface{}, fn ExecFn) error {
+func (m *metaClient) retryOnError(req interface{}, fn Fn) (interface{}, error) {
```

Comment:
How about ?

```go
type (
    ExecResp interface{
        GetErrorCode() nebula.ErrorCode 
        GetChangedLeader() *nebula.HostAddr
    }
    ExecFn func(req interface{}) (ExecResp, error)
)
func (m *metaClient) retryOnError(req interface{}, fn ExecFn) error {
    ...
}
You can set the `resp` in `ExecFn` to avoid type conversions?
```

Reply:
It's tedious,  each method must define a wrapper to implement ExecResp. I prefer using reflect to  get the response value.